### ISSUE TITLE
Symmetric 20% reproducibility

### ISSUE THREAD

**Issue by jinukim-ml** [lijichang/LNL-NCE#2]
Hi. First of all, I really want to say I am really inspired by your awesome work!

However, while reproducing the results from the paper I've encountered an issue.

With all the hyperparameters set same as the paper I can't reproduce the same result with 20% symmetric noise.

The paper says the best results is 96.2±0.09, but what I get consistently is around 95.2.

Is there something I missed? I would really appreciate your response, thanks!

**Comment by lijichang** [comment]
Dear,

Thank you for your interest in our project. 

In regards to the issues you've raised, would it be possible for you to provide us with more implementation details? Alternatively, you may also find it helpful to compare your implementation with the code we've provided.

Once again, thank you for your interest and we look forward to hearing from you soon.

Best regards,
Jichang

**Comment by jinukim-ml** [comment]
Thank you for the fast response!

I downloaded the code from this repository and ran the code as was with 3 different seeds.
I didn't change any hyperparameters because it was the same as the paper says.

1. Seed 123. Best: 95.09, last: 94.832
2. Seed 700. Best: 95.13, last: 94.983
3. Seed 999. Best: 95.57, last: 95.366

Other than that, the results are very consistent with the paper.
Let me know if I missed anything.

Thank you!

**Comment by changwxx** [comment]
I also find that Asymmetric 40% only achieves Best: 85.31, last: 84.76 (Seed 123). And I didn't change anything from the original code. Is there anything I missed? Many thanks in advance!

**Comment by lijichang** [comment]
@changwxx 
Thank you for your interest in our work.

I re-ran the experiments using both the original settings and different seed settings, and obtained results with an accuracy range of 92-94%. These results do not seem to have changed as significantly as you described.

For your question, could it be related to differences in environmental settings between our setups? Our code is based on DivideMix, and I also had difficulty reproducing their results initially. However, we were able to overcome this issue once we resolved some experimental setup problems.

**Comment by lijichang** [comment]
@Angry-Biscuit 

I'm sorry to forgot the reply of your question. Regarding your question, I have fixed it and updated the code.

**Comment by changwxx** [comment]
@lijichang 
Thank you for your quick response and helpful suggestions! 

I did not change anything except for the logger and my reproduction log shows as follows. All tasks work well as the paper reported except for Asymmetric 40%.

As you mentioned environmental settings, do you mean batch size, learning rate, or something? I would be appreciated if you could give me more instructions. Many thanks in advance!

![Snipaste_2023-04-12_15-43-35](https://user-images.githubusercontent.com/48717706/231388199-5515d957-5a85-49e3-89c2-cfb1436129f6.png)

**Comment by lijichang** [comment]
@changwxx 

I just downloaded this code onto my server and followed the README instructions to run my results. I didn't make any modifications. Of course, I reran the code based on my existing Python environment that were not re-installed newly.

**Comment by changwxx** [comment]
Sorry for the late response! Thank you for your kind help! I reran it 3 times and this bug seems disappeared for some reason. 
Now I can reproduce it with an accuracy about 94% :)
